Make sure that VDSO patch code is properly aligned

bjoernd opened the pull request:
The code that is injected into the tracee processes' VDSO needs to be 8 byte aligned in order to satisfy conditions imposed by the ptrace interface on 64 bit architectures. There have been reproducible reports of that not always being the case:
https://github.com/facebookexperimental/hermit/issues/41. Use an explicitly aligned data structure to enforce this.

bjoernd commented:
CLA: This is a contribution via Amazon, I am working with the internal team to get added to the CLA we have in place.

jasonwhite commented:
Great detective work! `aarch64` will likely have the same issue. Do you mind updating the same code for that architecture as well?

bjoernd commented:
Ack, will check AARCH64, too.

bjoernd commented:
Added the change for AARCH64 and renamed the data structure to a more generic `BufferAligned<size>`. ARM64 is only compile-tested because my use case (Hermit) doesn't even build on ARM64.

jasonwhite commented on reverie-ptrace/src/vdso.rs:
```diff
@@ -87,26 +100,36 @@ mod vdso_syms {
-    pub const clock_getres: &[u8; 16] = &[
+    #[repr(align(64))]
+    struct BufferAligned8([u8;8]);
+    struct BufferAligned16([u8;16]);
```
Shouldn't `BufferAligned16` have `#[repr(align(64))]` as well?

jasonwhite replied:
Hmm, maybe all your changes haven't been pushed, because I'm not seeing the more generic `BufferAligned<size>`.

bjoernd replied:
Indeed, sorry :)

jasonwhite reviewed (approved):
Looks good! Thanks for fixing this!
